stripe/_api_requestor.py

```diff
@@ -333,75 +388,97 @@ def request_raw(
         # makes these parameter strings easier to read.
         encoded_params = encoded_params.replace("%5B", "[").replace("%5D", "]")
 
+        encoded_body = encoded_params
+
+        supplied_headers = None
+        if (
+            "headers" in request_options
+            and request_options["headers"] is not None
+        ):
+            supplied_headers = dict(request_options["headers"])
+
+        headers = self.request_headers(method, api_mode, request_options)
+
         if method == "get" or method == "delete":
             if params:
                 abs_url = _build_api_url(abs_url, encoded_params)
             post_data = None
         elif method == "post":
-            if (
-                supplied_headers_dict is not None
-                and supplied_headers_dict.get("Content-Type")
-                == "multipart/form-data"
-            ):
+            if api_mode == "V1FILES":
                 generator = MultipartDataGenerator()
                 generator.add_params(params or {})
                 post_data = generator.get_post_data()
-                supplied_headers_dict[
+                headers[
```

Comment:
consider doing this in request_headers

Reply:
I considered it when implementing but since the `MultipartDataGenerator` needs params and the header value needs the generator's boundary value, I decided it made more sense to set the header here than in request headers.